### ISSUE TITLE
Depth Test Not Enabled Until First Image Blit

### ISSUE THREAD

**Issue by DiegoAce** [grimfang4/sdl-gpu#123]
In your 3D demo when GPU_SetDepthTest is enabled, GL_DEPTH_TEST is not enabled until the image is blitted. The first custom GL calls are not using the depth test. 

I attached a modified version of your 3D demo. The triangles and image are translated to see them overlap. Pressing SPACE blits the image which changes the depth test.

[main.c.zip](https://github.com/grimfang4/sdl-gpu/files/1811827/main.c.zip)

**Comment by DiegoAce** [comment]
If GL_DEPTH_TEST is manually enabled, it seems like GPU_ResetRendererState always disabled the depth test because last_depth_test is only set during a blit type function.

**Comment by DiegoAce** [comment]
Or perhaps that is not exactly the issue. I'm now trying to render all of my direct GL draws to a new target/framebuffer that has implemented GPU_AddDepthBuffer, but if I'm not using any GPU_Blit type functions on that target then the depth buffer isn't enabled. In this case manually calling glEnable(GL_DEPTH_TEST) before drawing to the target works. Then later calling GPU_ResetRendererState() allows for normal blitting on the main screen target. 
Perhaps there should be some type of function similar but opposite to GPU_ResetRendererState() that prepares a target for direct backend calls. That could set depth testing accordingly.

**Comment by grimfang4** [comment]
I want to look into this.  Are there any other details you've figured out?

**Comment by DiegoAce** [comment]
Not necessarily. I might be able to make some tests if you have a specific question in mind. 
In your 3D demo GL_DEPTH_TEST isn't attempted to be set from anywhere other than through GPU_ResetRendererState and GPU_Blit. So the very first direct backend draw is not using the depth test. 
I'm currently doing as I described in my previous comment, which is creating a new image and loading a target for it. Add a depth buffer. Call glEnable(GL_DEPTH_TEST). Make only direct backend draws to the target.  Reset renderer state. Then later blit this image to the main screen along with other stuff and all works fine. The image target is the only target I use the depth test for so if I never directly enable it it will never be enabled.

**Comment by DiegoAce** [comment]
For my personal project I realized I didn't need to create a new image/target just for the depth tested custom GL draws. I can draw them on the main screen as needed. I still need to call glEnable(GL_DEPTH_TEST) before though since my custom calls are the only ones I want depth tested.
